lib/recurly/errors.js

```diff
@@ -0,0 +1,298 @@
+import {EventDispatcher} from './event';
+
+const BASE_URL = 'https://dev.recurly.com/docs/recurly-js-';
+const ERRORS = [
+  {
+    code: 'not-configured',
+    message: 'Not configured. You must first call recurly.configure().',
+    classification: 'merchant',
+    help: 'getting-started#section-configure'
+  },
+  {
+    code: 'config-missing-public-key',
+    message: 'The publicKey setting is required.',
+    classification: 'merchant',
+    help: '#identify-your-site'
+  },
+  {
+    code: 'config-missing-fields',
+    message: 'The fields setting is required.',
+    classification: 'merchant'
+  },
+  {
+    code: 'missing-hosted-field-target',
+    message: c => `Target element not found for ${c.type} field using selector '${c.selector}'`,
+    classification: 'merchant'
+  },
+  {
+    code: 'api-error',
+    message: 'There was an error with your request.',
+    classification: 'metal'
```

Comment:
This is specific to Rails. Would something along the lines of `api`, `middleware`, or perhaps `service-layer` work?

Reply:
Sure, I'd agree that it's a bit overloaded. I think `api` is appropriate.